rust/arrow/src/compute/kernels/aggregate.rs

```diff
@@ -19,9 +19,42 @@
 
 use std::ops::Add;
 
-use crate::array::{Array, PrimitiveArray};
+use crate::array::{Array, LargeStringArray, PrimitiveArray, StringArray};
 use crate::datatypes::ArrowNumericType;
 
+/// Helper macro to perform min/max of strings
+macro_rules! min_max_string_helper {
+    ($array:expr, $cmp:tt) => {{
+        let null_count = $array.null_count();
+
+        if null_count == $array.len() {
+            return None
+        }
+        let mut n = "";
+        let mut has_value = false;
+        let data = $array.data();
+
+        if null_count == 0 {
+            for i in 0..data.len() {
+                let item = $array.value(i);
+                if !has_value || (&n $cmp &item) {
+                    has_value = true;
+                    n = item;
+                }
+            }
+        } else {
+            for i in 0..data.len() {
+                let item = $array.value(i);
+                if !has_value || data.is_valid(i) && (&n $cmp &item) {
```

Comment:
Is this correct? `&&` has higher precedence than `||` so I am reading this as `if !has_value || (data.is_valid(i) && (&n $cmp &item))` and since `!has_value` is true on the first iteration of the loop, it will always set `has_value = true` ?

Reply:
You are right. Amazing catch 💯💯.  I pushed a fix to it. This was also the case for primitive types, and so I also added a test and fix to them.